front/app/containers/Admin/reporting/components/ReportBuilder/TopBar/ReportTitle.tsx

```diff
@@ -73,23 +75,33 @@ const ReportTitle = ({ reportId }) => {
       if (reportTitleIsTaken(error)) {
         showError(formatMessage(messages.titleTaken));
       }
+
+      throw error;
     }
   };
 
-  const handleOnKeyDown = (e) => {
-    if (e.key === 'Enter') {
-      // To save the new title, we just remove focus from the input. The onBlur event
-      // will handle the rest. However, if the title is empty, enter does nothing.
-      //
-      // TODO: Ideally, we should find a way to keep the focus in the input if there is an error.
-      // However, I could not find a elegant way to do it that does not send multiple
-      // patch requests (one on enter, one on blur).
-      if (newTitle) inputRef?.blur();
+  const handleOnKeyDown = async (e) => {
+    if (e.key === 'Enter' && newTitle) {
+      try {
+        await updateTitle();
+
+        updateTitleOnBlurRef.current = false;
+        inputRef?.blur();
+        updateTitleOnBlurRef.current = true;
```

Comment:
❤️ I tried something similar with a state hook, but of course it didn't work out.